subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -807,14 +807,39 @@ static uint8_t ticker_resolve_collision(struct ticker_node *nodes,
 				(lazy_next - ticker_next->priority) >
 				(lazy_current - ticker->priority);
 
+			/* Does the current ticker with ticks_slot_window be
+			 * scheduled after the colliding ticker?
+			 */
+			uint8_t curr_has_ticks_slot_window =
+					(ticker->ext_data &&
+					 ticker->ext_data->ticks_slot_window &&
+					 ((acc_ticks_to_expire +
+					   ticker_next->ticks_slot) <
+					  (ticker->ext_data->ticks_slot_window -
+					   ticker->ticks_slot)));
+
+			/* Colliding next ticker does not use ticks_slot_window
+			 * or it does not fit after the current ticker within
+			 * the ticks_slot_window.
+			 */
+			uint8_t next_not_ticks_slot_window =
+					(!ticker_next->ext_data ||
+					 !ticker_next->ext_data->ticks_slot_window ||
+					 ((acc_ticks_to_expire +
+					   ticker_next->ext_data->ticks_slot_window -
+					   ticker_next->ticks_slot) <
+					  ticker->ticks_slot));
+
 			/* Check if next node is within this reservation slot
 			 * and wins conflict resolution
 			 */
-			if (!lazy_next_periodic_skip &&
-			    (next_force ||
-			     next_is_critical ||
-			     (next_has_priority && !current_is_older) ||
-			     (equal_priority && next_is_older))) {
+			if (curr_has_ticks_slot_window ||
+			    (!lazy_next_periodic_skip &&
+			     (next_force ||
+			      next_is_critical ||
+			      (next_has_priority && !current_is_older) ||
+			      (equal_priority && next_is_older &&
+			       next_not_ticks_slot_window)))) {
```

Comment:
The only thing I'm worried about here is the fact that with this change, ADV (or more precisely `ticks_slot_window`-nodes) become second class citizens. This means that they are more often skipped and placed at the end of a specific slot. This may remove the perturbation, and reduce the chance of being scanned by the peer. This happens in cases where the periodicity of advertising and e.q. an ACL connection coincide.

Reply:
Case 1: the current `ticker` be placed after the overlapping `ticker_next`, this is equivalent to `ticker` being placed after `ticker_next` when `random_delay` was applied. `ticker` could overlap with subsequent tickers, but that is a new collision resolution that takes into consideration parameters of current ticker with the subsequent ticker.

Case 2: the current `ticker` that is overlapping with the `ticker_next` had always been loosing to `ticker_next` irrespective of whether `ticker_next` used the `ticks_slot_window`. With the change in this PR, `ticker` would yield when `ticker_next` does not use `ticks_slot_window` or `ticker_next` cannot be placed after the current `ticker->ticks_slot` within `ticker_next`'s `ticks_slot_window`

With the changes in this PR, no regression in the existing connection related or mesh related bsim tests are uncovered. 

My belief being that this PR is not significantly changing runtime scheduling, and is assisting in placing the primary channel PDUs of extended advertising in between BIG events without cancelling BIG events.

Reply:
Within the available gap between connection events the advertiser event does have a random drift. For a single connection support usecase, the first advertisement does not have any change with this PR, but with multiple connection support, the connections have higher anchor point synchronization due to non-overlapping advertiser events. I think is an ok compromise.